source/agora/node/FullNode.d

```diff
@@ -1026,11 +1026,36 @@ public class FullNode : API
     {
         this.recordReq("postEnrollment");
 
+        /// FIXME: Use a proper type and sensible memory allocation pattern
+        version (all)
+        {
+            scope extraSet = new TestUTXOSet();
+            foreach (ref Hash hash, ref Transaction tx; this.pool)
+                extraSet.put(tx);
+            scope extraFinder = extraSet.getUTXOFinder();
```

Comment:
There's two main problems with this approach:
- We populate the `extraSet` with the content of the transaction pool even before we need it, so every `postEnrollment` will pay the price;
- We essentially duplicate the whole `pool` based on user input, which can create high delays and memory allocation;

Reply:
Ah, It's better to use `extraSet` when we fails to find a UTXO with the finder of the original utxo set. Ty.

Reply:
> We essentially duplicate the whole pool based on user input, which can create high delays and memory allocation;

Do you mean that we could maintain a future utxo set in the transaction pool?

Reply:
> > We essentially duplicate the whole pool based on user input, which can create high delays and memory allocation;
> 
> Do you mean that we could maintain a future utxo set in the transaction pool?

I don't think you meant that. Never mind.

Reply:
No I did not, but it's a potential solution, **iff** we only store the hashes of UTXO that are valid enrollment candidate (freeze txs with the right amount).

Reply:
Oh, I didn't think about it to that extent. Ty for the suggestion!

Reply:
Updated.